jupyterhubutils/lsstmgr/optionsformmanager.py

```diff
@@ -0,0 +1,146 @@
+'''Class to manage an LSST-specific options form.
+'''
+import datetime
+import jinja2
+import json
+from time import sleep
+from .. import SingletonScanner
+from ..utils import make_logger
+
+
+class LSSTOptionsFormManager(object):
+    '''Class to create and read a spawner form.
+    '''
+
+    _sizemap = {}
+    _sizelist = []
+    _scanner = None
+
+    def __init__(self, *args, **kwargs):
+        self.log = make_logger()
+        self.log.debug("Creating LSSTOptionsFormManager")
+        self.parent = kwargs.pop('parent')
+
+    def lsst_options_form(self):
+        '''Create an LSST Options Form from parent's config object.
+        '''
+        # Make options form by scanning container repository
+        cfg = self.parent.config
+        scanner = SingletonScanner(host=cfg.lab_repo_host,
+                                   owner=cfg.lab_repo_owner,
+                                   name=cfg.lab_repo_name,
+                                   experimentals=cfg.prepuller_experimentals,
+                                   dailies=cfg.prepuller_dailies,
+                                   weeklies=cfg.prepuller_weeklies,
+                                   releases=cfg.prepuller_releases,
+                                   cachefile=cfg.prepuller_cachefile,
+                                   debug=cfg.debug)
+        self._scanner = scanner
+        self._sync_scan()
+        lnames, ldescs = scanner.extract_image_info()
+        desclist = []
+        # If there's only one image, we don't need a form.
+        if not lnames or len(lnames) < 2:
+            return ""
```

Comment:
We do need a form so we can pick the size of our image.  Plus this gets rid of some call paths in make_pod_manifest

Reply:
**get_pod_manifest**